### PR TITLE
Implemented tests from server selection specification

### PR DESCRIPTION
**Pull request by rozza**
JAVA-1645

[Review on Reviewable](https://reviewable.io/reviews/rozza/mongo-java-driver/66)

### CODE REVIEW

File: driver-core/src/test/unit/com/mongodb/connection/JsonPoweredTest.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+            stringBuilder.append(line);
+            stringBuilder.append(ls);
+        }
+
+        return stringBuilder.toString();
+    }
+
+    static List<File> getTestFiles(final String resourcePath) throws URISyntaxException {
+        List<File> files = new ArrayList<File>();
+        addFilesFromDirectory(new File(ServerDiscoveryAndMonitoringTest.class.getResource(resourcePath).toURI()), files);
```
Comment (jyemin) [inline driver-core/src/test/unit/com/mongodb/connection/JsonPoweredTest.java]: Just for clarity, this should be JsonPoweredTest.class

**Comment by jyemin** [comment]
LGTM.

File: driver-core/src/test/unit/com/mongodb/connection/JsonPoweredTest.java (hunk 1 above)
Comment (rozza) [inline driver-core/src/test/unit/com/mongodb/connection/JsonPoweredTest.java]: Done.

**Comment by rozza** [comment]
Rebased and merged.

**Pull request closed without merging**